pkgs/development/beam-modules/mix-release.nix

```diff
@@ -0,0 +1,92 @@
+{ stdenv, lib, elixir, erlang, hex, rebar, rebar3, fetchMixDeps, makeWrapper, git }:
+
+{ pname
+, version
+, src
+, nativeBuildInputs ? [ ]
+, meta ? { }
+, enableDebugInfo ? false
+, mixEnv ? "prod"
+, compileFlags ? [ ]
+, mixDeps ? null
+, ...
+}@attrs:
+let
+  overridable = builtins.removeAttrs attrs [ "compileFlags" ];
+
+in
+stdenv.mkDerivation (overridable // {
+  dontStrip = true;
```

Comment:
Why do we not want to strip?

Reply:
My understanding is that stripping removes debugSymbols.
I've never needed to examine a crash on a packaged app but I feel that it's a better default to have those there.
I don't have any experience with how much stripping reduces binary size.
Personally binary size on elixir apps has never been a concern.
If somebody feels the need for it, we could make this overridable.

Reply:
> I don't have any experience with how much stripping reduces binary size.

Depending on the library it can be nothing to a few 100MBs. For example the java so's decreased by 100MB when stripping.

Reply:
Ok let me test with my app, post the results and ask other people's advice about it.

Reply:
Here are my results, without stripping here is 
```
❯ nix path-info -S ./result
/nix/store/fxzn5apn67lzd9gld9sgz82r3nmlk4n1-union-0.0.1   395793768
```
removing the `dontStrip = true`, here is what I get
```
/nix/store/qngs2vdzzj3claxbji2w96prrhgpjh1n-union-0.0.1   395791768
```
I can confirm the stripping ran since I have the additional line
```
strip is /nix/store/vvqjg5dk383chp7bsyhkkvxyv45yipfh-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/qngs2vdzzj3claxbji2w96prrhgpjh1n-union-0.0.1/lib  /nix/store/qngs2vdzzj3claxbji2w96prrhgpjh1n-union-0.0.1/bin
```

I have to say i don't understand why the stripping didn't work. Perhaps it's to do with the fact that the default stripper doesn't know how to strip a release binary.

I'm open if you have other suggestions, but otherwise I would say that we should leave it.

@SuperSandro2000 let me know.

Reply:
In any BEAM related release form, there is no need to do stripping.

BEAM modules can't/shouldn't be stripped after the fact, as that would invalidate their checksum if not done correctly, instead they should be compiled with appropriate settings before hand. At least `mix` would not include debug symbols by default in a production release.

Also the only ELF binaries and libraries that would get included and that could regular `strip` handle are exactly those from the buildtime (they are *copied*).

Either these are already stripped or they are passed in in their unstripped version on the packagers behalf. We should not override the packagers decission with an explicit flag that the packager needs to override to keep the explicitely activated debug symbols of the runtime.

Reply:
Thanks a lot for this!
Do you have experience with or know any good source talking about debugging a crash dump on a release? I'm interested (not in this PR) to add something in the docs about it, but it never happened to me. All in all I think the chances are fairly low that a crash dump would happen with the BEAM since it's designed against that, but just in case it happens to someone.

Reply:
BEAM crash dumps? NNo, I do not have any experience with recent versions of them, though back in OTP 16 they were basically roughly structured text files which contained a representation of the heap, stack and process dictionary of each and every process active at the time of the dump. As well as some process metadata.